File: matiasmasso/Mat.NET/DT2/DTOQualityDistribution.vb
Public Class DTOQualityDistribution
    Property Customer As DTOCustomer
    Property Skus As List(Of DTOProductSku)

    Public Sub New()
        MyBase.New
        _Skus = New List(Of DTOProductSku)
    End Sub

    Shared Function ExcelSheet(items As List(Of DTOQualityDistribution)) As MatHelperStd.ExcelHelper.Sheet
        Dim retval As New MatHelperStd.ExcelHelper.Sheet
        With retval
            .AddColumn("product depth", MatHelperStd.ExcelHelper.Sheet.NumberFormats.Integer)
            .AddColumn("customer")
        End With

        For Each item In items
            Dim oRow As New MatHelperStd.ExcelHelper.Row(retval)
            retval.Rows.Add(oRow)
            oRow.AddCell(item.Skus.Count)
            oRow.AddCell(item.Customer.FullNom)
        Next

        Return retval
    End Function

End Class

File: matiasmasso/Mat.NET/DTO/DTOQualityDistribution.cs
using System.Collections.Generic;

namespace DTO
{
    public class DTOQualityDistribution
    {
        public DTOCustomer Customer { get; set; }
        public List<DTOProductSku> Skus { get; set; }

        public DTOQualityDistribution() : base()
        {
            Skus = new List<DTOProductSku>();
        }

        public static MatHelper.Excel.Sheet ExcelSheet(List<DTOQualityDistribution> items)
        {
            MatHelper.Excel.Sheet retval = new MatHelper.Excel.Sheet();
            {
                var withBlock = retval;
                withBlock.AddColumn("product depth", MatHelper.Excel.Cell.NumberFormats.Integer);
                withBlock.AddColumn("customer");
            }

            foreach (var item in items)
            {
                var oRow = retval.AddRow();
                retval.Rows.Add(oRow);
                oRow.AddCell(item.Skus.Count);
                oRow.AddCell(item.Customer.FullNom);
            }

            return retval;
        }
    }

}
